tests/replication/_base.py

```diff
@@ -442,8 +443,11 @@ def connect_any_redis_attempts(self) -> None:
             self.assertEqual(host, "localhost")
             self.assertEqual(port, 6379)
 
-            client_protocol = client_factory.buildProtocol(None)
-            server_protocol = self._redis_server.buildProtocol(None)
+            client_address = IPv4Address("TCP", "127.0.0.1", 6379)
+            client_protocol = client_factory.buildProtocol(client_address)
+
+            server_address = IPv4Address("TCP", host, port)
+            server_protocol = self._redis_server.buildProtocol(server_address)
```

Comment:
I went for a different approach in https://github.com/matrix-org/synapse/pull/14988/commits/d9ed2d1283bf0989ec0844ed980cc7d1ef7511cb#diff-87257704840d75d10ae3941c55bfffff4aea0857c0d0aa7cecf36af5b6c4605dR157-R167 of passing in a dummy address. I should probably just go for something simpler like this.